### PR TITLE
Add GrantFund Deployer contract

### PR DESCRIPTION
**Pull request by prat-gpt**
# Description of change
## High level
* Add deployer contract to deploy grant fund, fund treasury, and startNewDistribution.

### CODE REVIEW

File: src/grants/Deployer.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import { GrantFund } from "./GrantFund.sol";
+
+contract Deployer {
+
+    GrantFund public grantFund;
+
+    function deployGrantFund(address ajnaToken_, uint256 treasury_) public returns (GrantFund) {
+
+        IERC20(ajnaToken_).transferFrom(msg.sender, address(this), treasury_);
```
Comment (grandizzy) [inline src/grants/Deployer.sol:26]: can move the transfer call down, after ajna token approve and right before `grantFund.fundTreasury` called?

File: src/grants/Deployer.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        grantFund = new GrantFund(ajnaToken_);
+
+        IERC20(ajnaToken_).approve(address(grantFund), treasury_);
+
+        grantFund.fundTreasury(treasury_);
+
+        grantFund.startNewDistributionPeriod();
+        return grantFund;
```
Comment (grandizzy) [inline src/grants/Deployer.sol]: would work with a named var `grantFund_` declared as returned by `deployGrantFund` function

**Review by grandizzy — COMMENTED** [review]
LGTM, minor comments added

**Comment by grandizzy** [comment]
Was also thinking if would make sense to have a set of require at the end, that is to assert that grant fund properly created / funded / distribution started (e.f. check treasury and contract balance, check distribution, others)

File: src/grants/Deployer.sol (hunk 1 above)
Comment (prat-gpt) [reply src/grants/Deployer.sol:26]: Updated in [d50335b](https://github.com/ajna-finance/ecosystem-coordination/pull/125/commits/d50335b997d90ad87f8b49378463e319d724628d)

File: src/grants/Deployer.sol (hunk 2 above)
Comment (prat-gpt) [reply src/grants/Deployer.sol]: Updated in [d50335b](https://github.com/ajna-finance/ecosystem-coordination/pull/125/commits/d50335b997d90ad87f8b49378463e319d724628d)

**Review by grandizzy — APPROVED** [review]
Thanks for changes, lgtm

**Review by EdNoepel — APPROVED** [review]
I look forward to testing this on Goerli.

File: test/unit/Deployer.t.sol [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        Deployer deployer = new Deployer();
+        ajnaToken.approve(address(deployer), treasury);
+
+        GrantFund grantFund = deployer.deployGrantFund(address(ajnaToken), treasury);
+
+        assertEq(grantFund.getDistributionId(), 1);
+
+        (,,,uint256 fundAvailable,,) = grantFund.getDistributionPeriodInfo(1);
+
+        assertEq(grantFund.treasury(), treasury - fundAvailable);
```
Comment (MikeHathaway) [inline test/unit/Deployer.t.sol:30]: Would be good to also check the fundsAvailable is .03 * treasury balance and > 0

Comment (prat-gpt) [reply test/unit/Deployer.t.sol:30]: Added in [92880a9](https://github.com/ajna-finance/ecosystem-coordination/pull/125/commits/92880a9bc73e48402a279719fda0152e324565c7)

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**